### PR TITLE
[Testing] Document improving test speed by reducing password encoder "work factor"

### PR DESCRIPTION
**Pull request by kbond**
I also reduced my test suite run time by pre-encoding passwords for my test fixtures. I'm not sure if we should document this.

### CODE REVIEW

**Comment by javiereguiluz** [comment]
Thanks for this contribution.

Question to @symfony/team-symfony-docs I'm 99.99% sure that we have or had something like this in the docs. Maybe in the "Best Practices"? But I can't find it. Can you remember? Did we remove this from docs for some reason? Thanks.

**Comment by xabbuh** [comment]
I thought we had something about the plaintext password encoder, but I couldn't find anything. What do you think if we change the example here to make use of the plaintext encoder?

**Comment by kbond** [comment]
A few arguments against plaintext: 

1. In my test suite, switching to the plaintext encoder breaks a few tests related to password migration (I check to see if a legacy passwords are properly migrated on login).
2. There was virtually no performance gain using plaintext vs sodium (with the lowest cost).

That being said, the plaintext config would be simpler. I'm fine switching if that's what you think would be best.

**Review by wouterj — APPROVED** [review]
I think this is great.

I also think we documented something like this already, but I think we must have lost it somewhere while moving things around (as I also can't find it anymore).

**Review by OskarStark — COMMENTED** [review]
Very nice, could you please add the comments to xml and php examples too? Thank you 👌🏻

File: testing/http_authentication.rst [hunk 1]
```diff
@@ -12,6 +12,65 @@ OAuth authentication services.
...
+        <?xml version="1.0" encoding="UTF-8"?>
+        <srv:container xmlns="http://symfony.com/schema/dic/security"
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xmlns:srv="http://symfony.com/schema/dic/services"
+            xsi:schemaLocation="http://symfony.com/schema/dic/services
+                https://symfony.com/schema/dic/services/services-1.0.xsd">
+
+            <config>
+                <encoder class="App\Entity\User" <!-- Use your user class name here -->
+                    algorithm="auto" <!-- This should be the same value as in config/packages/security.yaml -->
```
Comment (kbond) [inline testing/http_authentication.rst]: Comments aren't allow inside a tag - how should I handle?

**Comment by kbond** [comment]
> please add the comments to xml and php examples

Done. ~but see my issue with the xml comments.~

File: testing/http_authentication.rst (hunk 1 above)
Comment (wouterj) [reply testing/http_authentication.rst]: We usually put the comment above the tag and start with the attribute name:

```
<!-- algorithm: this should [...] -->
<encoder ...
```

Comment (kbond) [reply testing/http_authentication.rst]: Thanks!

**Comment by javiereguiluz** [comment]
Thanks a lot Kevin! This was finally merged in 4.4 and up.

**Pull request merged**